### PR TITLE
Heroku deployment of branches in pr

### PR DESCRIPTION
**Pull request by FaisalAbid**
Work in progress. Looking for feedback on approach.

See https://github.com/flutter/flutter/issues/8159

Currently failing to push to heroku due to an SSH key misconfiguration.

### CODE REVIEW

**Comment by abarth** [comment]
I think that request was for https://github.com/flutter/website rather than this repo.

**Comment by FaisalAbid** [comment]
Is that so? The issue was referencing code from this repo (docs being generated) , and the discussion was in this repo hence my PR here.

File: dev/bots/docs.sh [hunk 1]
```diff
@@ -14,8 +14,27 @@ FLUTTER_ROOT=$PWD dart dev/tools/javadoc.dart
 cp dev/docs/google2ed1af765c529f57.html dev/docs/doc
 
 # Upload new API docs when on Travis and branch is master
-
 if [ "$TRAVIS_PULL_REQUEST" = "false" ] && [ "$TRAVIS_BRANCH" = "master" ]; then
   cd dev/docs
   firebase deploy --project docs-flutter-io
+else
```
Comment (mit-mit) [inline dev/bots/docs.sh]: I don't think `else` will work here, as there are many configs travis can run in. For example, we don't want to run this on API jobs.

I think you need `[ "$TRAVIS_PULL_REQUEST" = "true" ] && [ "$TRAVIS_BRANCH" = "master" ];`

File: .travis.yml [hunk 2]
```diff
@@ -16,6 +18,8 @@ env:
 before_script:
   - ./dev/bots/travis_setup.sh
 script:
+  - export BRANCH=$(if [ "$TRAVIS_PULL_REQUEST" == "false" ]; then echo $TRAVIS_BRANCH; else echo $TRAVIS_PULL_REQUEST_BRANCH; fi)
```
Comment (mit-mit) [inline .travis.yml]: Is this some debugging content you left behind? Same for the next line.

Comment (mit-mit) [inline .travis.yml]: Can these lines be moved to the main script? Or are the variables needed not available there?

Comment (FaisalAbid) [inline .travis.yml]: These sure can, I felt it was cleaner to put them here. Will move them to the main script

File: dev/bots/docs.sh (hunk 1 above)
Comment (FaisalAbid) [inline dev/bots/docs.sh]: Correct, good point!

**Comment by abarth** [comment]
Further discussion on the bug indicates that this request was indeed for the website repo.  Sorry for the confusion.

**Pull request closed without merging**